src/main/java/teammates/common/datatransfer/FeedbackSessionResultsBundle.java

```diff
@@ -107,9 +107,11 @@ public FeedbackSessionResultsBundle(FeedbackSessionAttributes feedbackSession,
 
         // We change user email to team name here for display purposes.
         for (FeedbackResponseAttributes response : responses) {
-            if (questions.get(response.feedbackQuestionId).giverType == FeedbackParticipantType.TEAMS) {
-                // for TEAMS giver type, the giverEmail is stored as the student giver's email in the database
-                // but we convert it to the team name for use in FeedbackSessionResultsBundle
+            if (questions.get(response.feedbackQuestionId).giverType == FeedbackParticipantType.TEAMS
+                && roster.isStudentInCourse(response.giverEmail)) {
```

Comment:
Yup we should, I'll make a new issue for that and another issue to remove the extra condition and queries.